cloudcix_primitives/snapshot_lxd.py

```diff
@@ -1,31 +1,552 @@
 """
-Primitive for LXC Snapshot on LXD hosts
+POC: LXD Snapshot Management Module
+
+Example usage:
+# Create a snapshot with expiry date
+python3 snapshot_lxd.py build https://10.254.3.9:8443 default container02 containers snap00 false "2025-04-15T12:00:00Z" true
+
+# Read a snapshot
+python3 snapshot_lxd.py read https://10.254.3.9:8443 default container02 containers snap00 false false true
+
+# Read and restore from a snapshot (with stateful restore)
+python3 snapshot_lxd.py read https://10.254.3.9:8443 default container02 containers snap00 true true true
+
+# List all snapshots
+python3 snapshot_lxd.py read https://10.254.3.9:8443 default container02 containers none false false true
+
+# Delete a snapshot
+python3 snapshot_lxd.py scrub https://10.254.3.9:8443 default container02 containers snap00 true
+
+# Rename a snapshot
+python3 snapshot_lxd.py update https://10.254.3.9:8443 default container02 containers snap00 new_name none true
+
+# Update a snapshot's expiry date
+python3 snapshot_lxd.py update https://10.254.3.9:8443 default container02 containers snap00 none "2026-06-30T00:00:00Z" true
+
+# Update both name and expiry date
+python3 snapshot_lxd.py update https://10.254.3.9:8443 default container02 containers snap00 new_name "2026-06-30T00:00:00Z" true
 """
 # stdlib
-from typing import Tuple
-# lib
-# local
+from typing import Tuple, Optional
 
 
+# libs
+from cloudcix.rcc import API_SUCCESS, CHANNEL_SUCCESS, comms_lxd
+# local
+from cloudcix_primitives.utils import HostErrorFormatter, LXDCommsWrapper
+
 __all__ = [
     'build',
     'read',
     'scrub',
     'update',
 ]
 
+SUPPORTED_INSTANCES = ['containers']
+
+
+def build(
+        endpoint_url: str,
+        project: str,
+        instance_name: str,
+        instance_type: str,
+        snapshot_name: str,
+        stateful: bool = False,
+        expires_at: Optional[str] = None,
+        verify_lxd_certs: bool = True
+) -> Tuple[bool, str, dict]:
```

Comment:
```suggestion
) -> Tuple[bool, str]:
```

In the other primitives we return a bool and string only so we should be consistent.